aiopg/sa/result.py

```diff
@@ -333,27 +331,21 @@ def __iter__(self):
                       DeprecationWarning,
                       stacklevel=2)
         while True:
-            row = yield from self.fetchone()
+            row = yield from self.fetchone().__await__()
```

Comment:
This line worries me.
Drop `__iter__` for the sake of `__aiter__` maybe?

Reply:
I'm fine with dropping __iter__(), I tried to do this and a bunch of unit tests failed; they had to be modified to use `async for` instead of `for`, e.g.:
```diff
-    rows = [r for r in res]
+    rows = [r async for r in res]
```

So, this change will definitely break the API.  But since this was already raising a deprecation warning, I guess it's OK.